### PR TITLE
httomolib integration 

### PR DESCRIPTION
**Pull request by dkazanc**
Few things have been changed and added.
1. A wrapper for _image_ module that has been changed in httomolib. Tested with mpi.  
2. Some changes in the task_runner. @yousefmoazzam, I've generalised the wrapper for both tomopy and httomolib [here](https://github.com/DiamondLightSource/httomo/blob/0fc1f15c69c57c8f3e5e3605e992963e111120a9/src/httomo/task_runner.py#L444). Can you please check that it makes sense. I tested _image_ module and it works fine, will going to add more wrappers. 
3. Yaml generator for httomolib
4. generated templates

### CODE REVIEW

**Comment by namannimmo10** [comment]
> Not sure why is that though, may be @namannimmo10 knows?

I think it should work now: https://github.com/DiamondLightSource/httomolib/pull/32

File: src/httomo/task_runner.py [hunk 1]
```diff
@@ -441,26 +441,16 @@ def _run_method(func: Callable, task_no: int, package_name: str,
     # Add the appropriate dataset to the method function's dict of
     # parameters based on the parameter name for the method's python
     # function
-    if package_name == 'tomopy':
+    if (package_name == 'tomopy') or (package_name == 'httomolib'):
```
Comment (yousefmoazzam) [inline src/httomo/task_runner.py:444]: This section of code could be rejigged slightly to try and make it a bit clearer, my original code here was quite messy to begin with!

Seeing how it's changed, I'd suggest maybe something like this is clearer, what do you think:
```python
if (package_name == 'tomopy') or (package_name == 'httomolib'):
    httomo_params['data'] = datasets[in_dataset]

if method_name in savers_no_data_out_param:
    _run_method_wrapper(func, method_name, method_params, httomo_params)
    # Nothing more to do with output data if the saver has a special
    # kind of output
    return
else:
    # Run the method, then store the result in the appropriate
    # dataset in the `datasets` dict    
    datasets[out_dataset] = \
        _run_method_wrapper(func, method_name, method_params, httomo_params)
```

File: src/httomo/task_runner.py [hunk 2]
```diff
@@ -148,8 +148,8 @@ def run_tasks(
             # Check for any extra params unrelated to tomopy but related to
             # HTTomo that should be added in
             httomo_params = \
-                _check_signature_for_httomo_params(func, possible_extra_params)
-
+                _check_signature_for_httomo_params(func, possible_extra_params)
```
Comment (yousefmoazzam) [inline src/httomo/task_runner.py:151]: Looks like some random extra whitespace at the end of this line got added in? :stuck_out_tongue_closed_eyes:

**Review by yousefmoazzam — COMMENTED** [review]
Just a couple of comments, nothing major, all changes to the task runner look reasonable to me.

**Comment by yousefmoazzam** [comment]
I don't know how to comment on lines in a file that **haven't** been changed in the PR, seems like GitHub might still be working on that feature https://github.com/github/roadmap/issues/347? Anyway, I wanted to note that the `01_basic_cpu_pipeline_tomo_standard.yaml` pipeline YAML is broken since it still has `httomo.misc.corr.median_filter3d` in it which I'm not sure exists anymore in httomo? https://github.com/DiamondLightSource/httomo/blob/6061a46e069b61b03ca1e094016f73863780d15d/samples/pipeline_template_examples/01_basic_cpu_pipeline_tomo_standard.yaml#L12-L16

File: src/httomo/task_runner.py (hunk 1 above)
Comment (namannimmo10) [reply src/httomo/task_runner.py:444]:
```suggestion
    if package_name in ['httomolib', 'tomopy']:
```

File: src/httomo/task_runner.py [hunk 3]
```diff
@@ -441,26 +441,16 @@ def _run_method(func: Callable, task_no: int, package_name: str,
     # Add the appropriate dataset to the method function's dict of
     # parameters based on the parameter name for the method's python
     # function
-    if package_name == 'tomopy':
+    if (package_name == 'tomopy') or (package_name == 'httomolib'):
         httomo_params['data'] = datasets[in_dataset]
-    elif package_name == 'httomo':
```
[the remark this answers is not in the document]
Comment (dkazanc) [reply src/httomo/task_runner.py:446]: thanks for explanation  @yousefmoazzam. A fair point. In fact, I've been thinking what if we also provide an access to some function from skimage, such as [skimage.filters](https://scikit-image.org/docs/stable/api/skimage.filters.html#skimage.filters.farid) for instance. We have it as a dependency and there is a lot what experienced user might want to try. So this brings the problem you stated above. We could leave the condition for now if we decide to do this in future?

[the remark this answers is not in the document]
Comment (yousefmoazzam) [reply src/httomo/task_runner.py:446]: Ok, that makes sense to leave the possibility open, so lets keep the `_set_method_data_param()` function.

For each package being wrapped there'd need to be a way of taking care of its "standard name" for the input data parameter.

For example, that scikit-image function you linked has `image` as the function's data parameter, so something like the following would be needed to handle that:
```python
if package_name in ['httomolib', 'tomopy']:
    httomo_params['data'] = datasets[in_dataset]
elif package_name == 'skimage':
    httomo_params['image'] = datasets[in_dataset]
```

And then a similar `elif` block would be added for each package with a different "standard name" for the data parameter.

**Comment by yousefmoazzam** [comment]
On a similar note, the `httomolib.misc.images.save_to_images` config in `01_basic_cpu_pipeline_tomo_standard.yaml` should be changed too I think, currently it's also broken as https://github.com/DiamondLightSource/httomo/blob/6061a46e069b61b03ca1e094016f73863780d15d/samples/pipeline_template_examples/01_basic_cpu_pipeline_tomo_standard.yaml#L54-L61 and making it the same as `02_basic_cpu_pipeline_tomo_standard.yaml` fixes it https://github.com/DiamondLightSource/httomo/blob/7e796ebdace1102dfee49ab3ca29c376f940f99b/samples/pipeline_template_examples/02_basic_cpu_pipeline_tomo_standard.yaml#L55-L64

File: src/httomo/task_runner.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (dkazanc) [reply src/httomo/task_runner.py:446]: sounds good, thanks!

**Pull request merged**